src/coap_tinydtls.c

```diff
@@ -336,6 +336,8 @@ coap_dtls_free_session(coap_session_t *coap_session) {
     coap_log(LOG_DEBUG, "***removed session %p\n", coap_session->tls);
     coap_free_type(COAP_DTLS_SESSION, coap_session->tls);
     coap_session->tls = NULL;
+    if (coap_session->context)
```

Comment:
Given that `coap_session->context` previously has been used in this function, this check is redundant, or should be moved to the beginning, respectively.

Reply:
Code reworked to protect usage of ctx and coap_session->context